### PR TITLE
RS-325: How to Store Vibration Sensor Data | Part 2

### PR DESCRIPTION
**Pull request by AnthonyCvn**

### CODE REVIEW

File: blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+image: ./img/vibration_data_flow.jpg
+---
+
+import FilterExampleImg from './img/filter-example.webp';
+import SettingsExampleImg from './img/settings-example.webp';
+import DashboardExampleImg from './img/dashboard-example.webp';
+
+![Vibration Data Flow](./img/vibration_data_flow.jpg)
+
+**[ReductStore](/)** is designed to efficiently handle time series object data, making it an excellent choice for storing high frequency vibration sensor measurements.
```
Comment (atimin) [inline blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx:17]: I'm not sure if object data exists. blob, binary or unstructured data.

File: blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+):
+    labels = {
+        "rms": "high" if rms > HIGH_RMS else "low",
+        "peak_to_peak": "high" if peak_to_peak > HIGH_PEAK_TO_PEAK else "low",
+        "crest_factor": "high" if crest_factor > HIGH_CREST_FACTOR else "low",
+    }
+    await bucket.write("sensor_readings", packed_data, timestamp, labels=labels)
+```
+
+This is where we store our packed data, along with labels that indicate whether each metric is high or low.
```
Comment (atimin) [inline blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx:168]: data is too abstract and someone could not get that we actually write  a chunk. Lets say packed `chunk of data`.

File: blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        signal = struct.unpack(fmt, data)
+        signal = np.array(signal, dtype=np.float32)
+
+        print(f"Number of data points: {num_points}")
+        print(f"First few values: {signal[:5]}")
+        print("---")
+```
+
+This function shows how to query data within a given time range and unpack the binary data back into the original Numpy array.
+1. The `read_all()` method reads all binary data at a given timestamp.
```
Comment (atimin) [inline blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx:196]: read the whole chunk of data

File: blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+    # Query the stored data for the last 5 seconds
+    end_time = int(time.time() * 1_000_000)
+    start_time = end_time - 5_000_000
+    await query_data(bucket, start_time, end_time)
+```
+
+This is the main execution flow of our script, which demonstrates the complete data flow:
+1. We connect to ReductStore and create a bucket.
+2. We store 5 seconds of data, with 1 second of data stored every second. Each timestamp is generated using the current time in microseconds.
```
Comment (atimin) [inline blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx:227]: in chunks of 1 second data.

File: blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+## Replication with ReductStore Web Console
+
+In addition to storing and querying data, we can also set up replication to duplicate our sensor data across multiple ReductStore instances.
+
+Replication tasks can be configured in a variety of ways, such as
+- Replicate all data from a source bucket to a target bucket (e.g. `sensor_data` to `backup_data`)
+- Replicate only specific entries, e.g. `sensor_readings`.
+- Replicate data based on labels, e.g. replicate only data with `high` RMS values to `high_peak_to_peak` bucket (as we'll do in this example).
+
+The replication task can be set using client libraries, HTTP API, provisioning, or the ReductStore web console.
```
Comment (atimin) [inline blog/2024-07-12-data-flow-with-vibration-data-part-2/index.mdx:241]: and CLI client

**Review by atimin — CHANGES_REQUESTED** [review]
Very good, I just have few suggestion.

**Comment by AnthonyCvn** [comment]
@atimin thanks for the review 👍  I made the changes and will merge.

**Pull request merged**